examples/nirfmxwlan/sem.py

```diff
@@ -0,0 +1,359 @@
+    upper_offset_margin_absolute_power = (
+        sem_fetch_upper_offset_margin_array_response.margin_absolute_power
+    )
+    upper_offset_margin_relative_power = (
+        sem_fetch_upper_offset_margin_array_response.margin_relative_power
+    )
+
+    sem_fetch_spectrum_response = client.SEMFetchSpectrum(
+        nirfmxwlan_types.SEMFetchSpectrumRequest(
+            instrument=instr,
+            selector_string="",
+            timeout=10.0,
+        )
+    )
+    check_for_warning(sem_fetch_spectrum_response, instr)
+
+    print("\n----------SEM Results----------\n")
+    if measurement_status == nirfmxwlan_types.SEM_MEASUREMENT_STATUS_PASS:
+        print("Measurement Status               : PASS")
+    elif measurement_status == nirfmxwlan_types.SEM_MEASUREMENT_STATUS_FAIL:
+        print("Measurement Status               : FAIL")
+    print(f"Carrier Absolute Power (dBm)     : {absolute_power}")
+    print("\n-----------Lower Offset Measurements----------- \n")
+    for i in range(lower_offset_array_size):
+        print(f"Offset  {i}")
+        if (
+            lower_offset_measurement_status[i]
+            == nirfmxwlan_types.SEM_LOWER_OFFSET_MEASUREMENT_STATUS_PASS
+        ):
+            print("Measurement Status               : PASS")
+        elif (
+            lower_offset_measurement_status[i]
+            == nirfmxwlan_types.SEM_LOWER_OFFSET_MEASUREMENT_STATUS_FAIL
+        ):
+            print("Measurement Status               : FAIL")
+        print(f"Margin (dB)                      : {lower_offset_margin[i]}")
+        print(f"Margin Frequency (Hz)            : {lower_offset_margin_frequency[i]}")
+        print(f"Margin Absolute Power (dBm)      : {lower_offset_margin_absolute_power[i]}")
+    print("\n-----------Upper Offset Measurements----------- \n")
+    for i in range(upper_offset_array_size):
+        print(f"Offset  {i}")
+        if (
+            upper_offset_measurement_status[i]
+            == nirfmxwlan_types.SEM_UPPER_OFFSET_MEASUREMENT_STATUS_PASS
+        ):
+            print("Measurement Status               : PASS")
+        elif (
+            upper_offset_measurement_status[i]
+            == nirfmxwlan_types.SEM_UPPER_OFFSET_MEASUREMENT_STATUS_FAIL
+        ):
+            print("Measurement Status               : FAIL")
+        print(f"Margin (dB)                      : {upper_offset_margin[i]}")
+        print(f"Margin Frequency (Hz)            : {upper_offset_margin_frequency[i]}")
+        print(f"Margin Absolute Power (dBm)      : {upper_offset_margin_absolute_power[i]}")
+
+except grpc.RpcError as rpc_error:
+    error_message = rpc_error.details()
+    for entry in rpc_error.trailing_metadata() or []:
+        if entry.key == "ni-error":
+            value = entry.value if isinstance(entry.value, str) else entry.value.decode("utf-8")
+            error_message += f"\nError status: {value}"
```

Comment:
```
examples/nirfmxwlan/sem.py:349: error: Unsupported left operand type for + ("None")  [operator]
examples/nirfmxwlan/sem.py:349: note: Left operand is of type "Optional[str]"
```

`rpc_error.details()` is `Optional[str]`, a.k.a `str | None`: https://github.com/shabbyrobe/grpc-stubs/blob/master/grpc-stubs/__init__.pyi#L428

The most concise solution is to do `error_message = rpc_error.details() or ""`. Adding an assertion would also work.